### ISSUE TITLE
assert.ObjectsAreEqual work not correct with serialized\deserialized time.Time

### ISSUE THREAD

**Issue by senyast4745** [stretchr/testify#950]
**time.Time** not equals before and after serialization/deserialization to JSON on machine in **UTC**
reason: https://github.com/golang/go/issues/19486#issuecomment-292610767

code: https://play.golang.org/p/H0ACWIBZE3O

solution: change the **time.Time** comparison format. For example, compare time using **time.Equal**

**Comment by senyast4745** [comment]
when using **reflect.DeepEqual**, **time.Time** before and after serialization \ deserialization are not equal

**Comment by ryepup** [comment]
FYI, this bug is back; the fix got reverted for some reason on ed4976c764638c2061fa1e952539ecaf75ea97f8

**Comment by fasmat** [comment]
@senyast4745 @boyan-soubachov I'm confused by this. The fix was merged on Jul 27, 2020 and immediately reverted again. As of today the following code snipped still fails to assert that the timestamps are equal:

```go
func TestMarshal(t *testing.T) {
	expected := time.Now()

	js, err := json.Marshal(expected)
	require.NoError(t, err)

	var actual time.Time
	require.NoError(t, json.Unmarshal(js, &actual))

	assert.Equal(t, expected, actual)
}
```

Output:

```
--- FAIL: TestMarshal (0.00s)
    /workspaces/spacemesh/go-spacemesh/activation/wire/wire_v1_test.go:25: 
        	Error Trace:	/workspaces/spacemesh/go-spacemesh/activation/wire/wire_v1_test.go:25
        	Error:      	Not equal: 
        	            	expected: time.Date(2025, time.January, 2, 15, 59, 30, 200470671, time.Local)
        	            	actual  : time.Date(2025, time.January, 2, 15, 59, 30, 200470671, time.UTC)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-(time.Time) 2025-01-02 15:59:30.200470671 +0000 UTC m=+0.004242334
        	            	+(time.Time) 2025-01-02 15:59:30.200470671 +0000 UTC
        	            	 
        	Test:       	TestMarshal
```

Should this issue be re-opened?
